lib/jwt/jwk/ec.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+module JWT
+  module JWK
+    class EC
+      extend Forwardable
+      def_delegators :@keypair, :private?, :public_key
+
+      attr_reader :keypair
+      attr_reader :kid
+
+      KTY    = 'EC'.freeze
+      BINARY = 2
+
+      def initialize(keypair, kid = nil)
+        raise ArgumentError, 'keypair must be of type OpenSSL::PKey::EC' unless keypair.is_a?(OpenSSL::PKey::EC)
+
+        @keypair = keypair
+        @kid = kid || generate_kid(@keypair)
+      end
+
+      def export
+        crv, x_octets, y_octets = keypair_components(keypair)
+        {
```

Comment:
Export private key if option `include_private` is set to true like HMAC JWK does: https://github.com/jwt/ruby-jwt/blob/c5bf3376a6ff8bd619bd724815d9e8e6bfc3accd/lib/jwt/jwk/hmac.rb#L30